thrift/transport.go

```diff
@@ -49,6 +54,37 @@ func (t *readWriterTransport) Close() error {
 	return nil
 }
 
+func (t *readWriterTransport) ReadByte() (byte, error) {
+	v := t.readBuf[0:1]
+
+	var n int
+	var err error
+
+	for {
+		n, err = t.Read(v)
```

Comment:
Since we wrap any `Reader`, it is possible according to the `Reader` interface:
https://golang.org/pkg/io/#Reader

Specifically,
```
Implementations of Read are discouraged from returning a zero byte count with a nil error, except when len(p) == 0. Callers should treat a return of 0 and nil as indicating that nothing happened; in particular it does not indicate EOF.
```
So implementations are discouraged from doing that, but callers should ignore 0 and nil.